app/celery/tasks.py

```diff
@@ -83,15 +88,42 @@ def process_job(job_id, sender_id=None, shatter_batch_size=DEFAULT_SHATTER_JOB_R
             get_id_task_args_kwargs_for_job_row(row, template, job, service, sender_id=sender_id)[1]
             for row in shatter_batch
         ]
+        _shatter_job_rows_with_subdivision(template.template_type, batch_args_kwargs)
+
+    job_complete(job, start=start)
+
+
+def _shatter_job_rows_with_subdivision(template_type, args_kwargs_seq, top_level=True):
+    try:
         shatter_job_rows.apply_async(
             (
-                template.template_type,
-                batch_args_kwargs,
+                template_type,
+                args_kwargs_seq,
             ),
             queue=QueueNames.JOBS,
         )
+        if not top_level:
+            log_context = {"shatter_batch_size": len(args_kwargs_seq)}
+            current_app.logger.info(
+                "Job shatter batch sent with reduced size of %(shatter_batch_size)s", log_context, extra=log_context
+            )
+    except BotoClientError as e:
```

Comment:
Very optional because I’m not sure if this is actually more readable or not, but consider:
```suggestion
    except BotoClientError as e:
        ...
    else:
        if not top_level:
            log_context = {"shatter_batch_size": len(args_kwargs_seq)}
            current_app.logger.info(
                "Job shatter batch sent with reduced size of %(shatter_batch_size)s", log_context, extra=log_context
            )
```

Reply:
Yeah maybe